modules/mediagoBidAdapter.js

```diff
@@ -0,0 +1,458 @@
+}
+
+/**
+ * 将尺寸转为RTB识别的尺寸
+ *
+ * @param  {Array|Object} requestSizes 配置尺寸
+ * @return {Object}
+ */
+function transformSizes(requestSizes) {
+  let sizes = [];
+  let sizeObj = {};
+
+  if (
+    utils.isArray(requestSizes) &&
+    requestSizes.length === 2 &&
+    !utils.isArray(requestSizes[0])
+  ) {
+    sizeObj.width = parseInt(requestSizes[0], 10);
+    sizeObj.height = parseInt(requestSizes[1], 10);
+    sizes.push(sizeObj);
+  } else if (typeof requestSizes === 'object') {
+    for (let i = 0; i < requestSizes.length; i++) {
+      let size = requestSizes[i];
+      sizeObj = {};
+      sizeObj.width = parseInt(size[0], 10);
+      sizeObj.height = parseInt(size[1], 10);
+      sizes.push(sizeObj);
+    }
+  }
+
+  return sizes;
+}
+
+// 支持的广告尺寸
+const mediagoAdSize = [
+  { w: 300, h: 250 },
+  { w: 300, h: 600 },
+  { w: 728, h: 90 },
+  { w: 970, h: 250 },
+  { w: 320, h: 50 },
+  { w: 160, h: 600 },
+  { w: 320, h: 180 },
+  { w: 320, h: 100 },
+  { w: 336, h: 280 },
+];
+
+/**
+ * 获取广告位配置
+ * @param {Array}  validBidRequests an an array of bids
+ * @param {Object} bidderRequest  The master bidRequest object
+ * @return {Object}
+ */
+function getItems(validBidRequests, bidderRequest) {
+  let items = [];
+  items = validBidRequests.map((req, i) => {
+    let ret = {};
+    let mediaTypes = getProperty(req, 'mediaTypes');
+
+    let sizes = transformSizes(getProperty(req, 'sizes'));
+    let matchSize;
+
+    // 确认尺寸是否符合我们要求
+    for (let size of sizes) {
+      matchSize = mediagoAdSize.find(
+        (item) => size.width === item.w && size.height === item.h
+      );
+      if (matchSize) {
+        break;
+      }
+    }
+    if (!matchSize) {
+      return {};
+    }
+
+    const bidFloor = getBidFloor(req);
+    // const gpid =
+    //   utils.deepAccess(req, 'ortb2Imp.ext.gpid') ||
+    //   utils.deepAccess(req, 'ortb2Imp.ext.data.pbadslot') ||
+    //   utils.deepAccess(req, 'params.placementId', 0);
+    // console.log("wjh getItems:", req, bidFloor, gpid);
+
+    // if (mediaTypes.native) {}
+    // banner广告类型
+    if (mediaTypes.banner) {
+      let id = '' + (i + 1);
+      ret = {
+        id: id,
+        bidfloor: bidFloor,
+        banner: {
+          h: matchSize.h,
+          w: matchSize.w,
+          pos: 1,
+        },
+        ext: {
+        //   gpid: gpid, // 加入后无法返回广告
+        },
+      };
+      itemMaps[id] = {
+        req,
+        ret,
+      };
+    }
+
+    return ret;
+  });
+  return items;
+}
+
+/**
+ * 获取rtb请求参数
+ *
+ * @param {Array}  validBidRequests an an array of bids
+ * @param {Object} bidderRequest  The master bidRequest object
+ * @return {Object}
+ */
+function getParam(validBidRequests, bidderRequest) {
+  const pubcid = utils.deepAccess(validBidRequests[0], 'crumbs.pubcid');
+  // console.log('wjh getParam', validBidRequests, bidderRequest);
+  let isMobile = isMobileAndTablet() ? 1 : 0;
+  let isTest = 0;
+  let auctionId = getProperty(bidderRequest, 'auctionId') || getRandomId();
```

Comment:
I don't see a reason to have getRandomId. There should never be a scenario where a bidRequest does not have an auctionId here. Is there a reason you decided to create getRandomId for this?

Reply:
All right,I will remove this code.